src/systemcmds/topic_listener/CMakeLists.txt

```diff
@@ -32,7 +32,7 @@
 ############################################################################
 
 add_custom_command(OUTPUT topic_listener.cpp
-                COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/generate_listener.py ${PX4_SOURCE_DIR} ${EXTERNAL_MODULES_LOCATION} > topic_listener.cpp
```

Comment:
Why remove EXTERNAL_MODULES_LOCATION?

Reply:
It is not necessary. As now you can set any sub-directory under msg/,   like this:

msg/
       abc.msg
       cde.msg
       external_location/
                        aaa.msg
                        bbb.msg
       other_subdir/
                        ccc.msg

and in msg/CMakeList.txt:

set (msg_files
     ....
      abc.msg
      external_location/aaa.msg
      external_location/bbb.msg
     ....
}

much more flexible for topic expansion. Having an extra EXTERNAL_MODULES_LOCATION param seems redundant.

Reply:
External modules is a mechanism for having the entire thing live outside of the tree and get into the build without making changes (environment variable set at configure time).

Reply:
I see.
In that case, will it better to manage to merge the external paths in parent level CMakeFile with $(msg_files) before pass it to py script?

Reply:
I believe calling the python script separately would work fine. It's slower to do this for every single .msg file, but doing it in a few groups (msg/, external modules, other_subdir) should be fine.